File: 프로그래머스/lv3/43238. 입국심사/입국심사.py
def solution(n, times):
    left, right = min(times), max(times)*n
    answer = max(times)*n
    
    while left <= right:
        total = 0
        mid = (left + right) // 2
        for time in times:
            total += (mid // time)
        if total < n:
            left = mid + 1
        else:
            right = mid - 1
            answer = min(answer, mid)
            
    return answer

File: 프로그래머스/lv3/43238. 입국심사/입국심사.js
function solution(n, times) {
    times.sort((a,b)=> a-b);
    var left = 1;
    var right = n*times[times.length - 1];
    var answer = right;
    
    while (left <= right) {
        var mid = Math.floor((left+right)/2);
        var cnt = 0;
        for (time of times) {
            cnt += Math.floor(mid/time);
            if (cnt >= n) {
                answer = Math.min(mid, answer);
            }
        }
        if (cnt>=n) right = mid-1;
        else left = mid+1;
    }
    return answer;
}